eth/api_backend.go

```diff
@@ -263,7 +263,8 @@ func (b *EthAPIBackend) GetEVM(ctx context.Context, msg *core.Message, state *st
 	if blockCtx != nil {
 		context = *blockCtx
 	} else {
-		context = core.NewEVMBlockContext(header, b.eth.BlockChain(), nil, b.eth.blockchain.Config(), state)
+		feeCurrencyContext := core.GetFeeCurrencyContext(header, b.eth.blockchain.Config(), state)
+		context = core.NewEVMBlockContext(header, b.eth.BlockChain(), nil, b.eth.blockchain.Config(), state, feeCurrencyContext)
```

Comment:
We have plenty of changes like this where we don't want any change in behaviour compared to before this PR. We could avoid that by keeping the old `NewEVMBlockContext` and having a new `NewEVMBlockContextFromFeeCurrencyContext` that we use where we intent a different behaviour than before.

Pro:
- Smaller diff

Con:
- Makes exactly the kind of bug we want to solve more likely, because the compile won't ask you to look at every place where a new context is creatd.

Reply:
Yes, I agree.
I forced that declaration change due the `con`. I prefer to hate a little more the merge than miss this context error